xrviz/dashboard.py

```diff
@@ -145,4 +166,75 @@ def create_indexed_graph(self):
         sel_data = sel_data.sel(**selection, drop=True)
         assign_opts = {dim: self.data[dim] for dim in sel_data.dims}
         graph = sel_data.assign_coords(**assign_opts).hvplot.quadmesh(**graph_opts)
-        return graph
+        self.output[0] = graph
+
+    def create_selectors_players(self, graph):
+        """
+        This function is applicable for non_indexed coords, in case
+        sliders are generated. This function moves the sliders to bottom of
+        graph if they are present and convert them into Selectors,Players.
+        """
+        graph = pn.Row(graph)
+        try:  # `if graph[0][1]` or `len(graph[0][1])` results in error in case it is not present
+            if graph[0][1]:  # if sliders are generated
+                self.output[0] = graph[0][0]
+
+                # link the generated slider with agg selector in fields 
+                for slider in graph[0][1]:
+                    for dim in self.kwargs['dims_to_select_animate']:
+                        if slider.name == dim or slider.name == self.data[dim].long_name:
+                            if self.kwargs[dim] == 'Select':
+                                selector = convert_widget(slider, pn.widgets.Select())
+                            else:
+                                selector = convert_widget(slider, pn.widgets.DiscretePlayer())
+                            self.index_selectors.append(selector)
+
+                for selector in self.index_selectors:
+                    if isinstance(selector, pn.widgets.Select):
+                        self.output[1].append(selector)
+                    else:
+                        player = player_with_name_and_value(selector)
+                        self.output[1].append(player)
+
+        except:  # else return simple graph
+            self.output[0] = graph
+
+    def is_non_indexed_coord(self, x):
+        indexed_coords = set(self.data[self.var].dims) & set(self.data[self.var].coords)
+        non_indexed_coords = list(set(self.data[self.var].coords) - indexed_coords)
+        if x in non_indexed_coords:
+            return True
+        else:
+            return False
+
+    def set_data(self, data):
+        if isinstance(data, xr.Dataset):
+            self.data = data
+            self.is_dataset = True
+        elif isinstance(data, xr.DataArray):
+            self.data = data
+            self.is_dataset = False
+        else:
+            raise ValueError
+
+    def set_coords(self, *args):
+        # We can't reset indexed coordinates so add them every time
+        # in coord_selector.value
+        self.data = self.data.reset_coords()
+        indexed_coords = set(self.data.dims) & set(self.data.coords)
```

Comment:
Maybe cleaner to call `set.intersection`

Reply:
Corrected. Thanks!